app/models/mixins/authentication_mixin.rb

```diff
@@ -2,10 +2,49 @@ module AuthenticationMixin
   extend ActiveSupport::Concern
 
   included do
+    # There are some dirty, dirty Ruby/Rails reasons why this method needs to
+    # exist like this, and I will try and explain:
+    #
+    # First, we need the extra +.where+ here because if this is used in a
+    # nested SELECT statement (aka: virtual_delegate), the +resource_type+
+    # check is dropped on the floor and not included in the subquery.  As a
+    # result, it will just pickup the first record to match the
+    # relationship_id column and that could be associated with any OTHER object
+    # with an Authentication record.
+    #
+    # For example, you get back an authentication record for an EMS record when
+    # you are looking up on for a Host.
+    #
+    # Secondly, the reason this method exists, and is FIRST (prior to the
+    # +has_one+ that uses it) is it needs to be defined prior to the
+    # relationship method that calls it.  Also, it needs to be a method so the
+    # proper local variable, +authentication_mixin_relation+, can be defined
+    # with the class that the +resource_type+ needs to match and remain in
+    # scope for the Proc below.  If it is a class variable or done in other
+    # fashion, it will be overwritten whenever this module is included
+    # elsewhere.
```

Comment:
I don't understand this (which may be why my comments below sound ignorant).  `name` will be the included class.  I'm not sure why the proc is required.

Reply:
The `proc` was always there, it was just written as this previously:

```ruby
-> { order(Authentication::STATUS_SEVERITY_AREL.desc) }
```

Reply:
Sorry, answered just this bit:

>  I'm not sure why the proc is required ...

I answered this better here:  https://github.com/ManageIQ/manageiq/pull/20494#discussion_r478621643

(with examples)